postprocessing/Consumer.py

```diff
@@ -189,39 +150,121 @@ def update_processes(self):
                         self.instrument_jobs[instrument].remove(i)
                 self.procList.remove(i)
 
-    def heartbeat(self, destination=None, data_dict={}):
-        """
-        Send heartbeats at a regular time interval
-        @param: destination where to send the heartbeat
-        @param data_dict: optional dictionary to pass along
-        """
-        try:
-            if destination is None:
-                destination = self.config.heart_beat
-            stomp = sync.Stomp(self.stompConfig)
-            stomp.connect()
-            if not type(data_dict) == dict:
-                logging.error("Heartbeat argument data_dict was not a dict")
-                data_dict = {}
-            data_dict.update(
-                {
-                    "src_name": socket.gethostname(),
-                    "role": "postprocessing",
-                    "status": "0",
-                    "pid": str(os.getpid()),
-                }
-            )
-            stomp.send(destination, json.dumps(data_dict).encode())
-            stomp.disconnect()
-        except:
-            logging.error("Could not send heartbeat: %s" % sys.exc_info()[1])
-
     def ack_ping(self, data):
         """
         Send an ACK message in response to a ping
         @param data: data that was received with the ping request
         """
         if "reply_to" in data:
-            self.heartbeat(data["reply_to"], data)
+            heartbeat(self.conn, data["reply_to"], data)
         else:
             logging.error("Incomplete ping request %s" % str(data))
+
+
+def heartbeat(conn, destination, data_dict={}):
+    """
+    Send heartbeats at a regular time interval
+    @param: destination where to send the heartbeat
+    @param data_dict: optional dictionary to pass along
+    """
+
+    try:
+        if not isinstance(data_dict, dict):
+            logging.error("Heartbeat argument data_dict was not a dict")
+            data_dict = {}
+        data_dict.update(
+            {
+                "src_name": socket.gethostname(),
+                "role": "postprocessing",
+                "status": "0",
+                "pid": str(os.getpid()),
+            }
+        )
+        conn.send(destination, json.dumps(data_dict).encode())
+    except:
+        logging.error("Could not send heartbeat: %s" % sys.exc_info()[1])
```

Comment:
It may be better to change this to
```suggestion
        logging.exception("Could not send heartbeat")
```